yarn-completion.bash

```diff
@@ -902,7 +908,11 @@ _yarn_workspace() {
 					declare workspace_path
 					declare -a workspaces=()
 					for workspace_path in $(__yarn_get_package_fields -t array workspaces); do
-						workspaces+=("$(basename "$workspace_path")")
+						declare cwd=$PWD
+						until [[ -f "$cwd/package.json" ]]; do
```

Comment:
This is dangerous. I wouldn't use an until or a while without something that is known for sure to absolutely exit no matter what.

Reply:
See line 36. It will break if it traverses to all the way to the system root directory without locating a `package.json`.

But.... it might be better to instead break when `"$cwd"` equals the original `"$cwd"`, because then that would mean we're back at the project root.